docs/concepts/sync-vs-async.rst

```diff
@@ -4,36 +4,54 @@ Sync vs Async Applications
 
 TL;DR
 ------
-1. For applications with horizontal scaling: multiple users, lots of i/o throughput,... go with async. Check out:
+
+Burr supports synchronous (standard python) and asynchronous (``async def``/``await``) applications. At a high leveL:
+
+1. Use the `async` interfaces when you have I/O-heavy applications that require horizontal scaling, and have avaialble asynchronous APIs (E.G. async LLM APIs)
 
    * :py:meth:`.abuild() <.ApplicationBuilder.abuild()>`
    * :py:meth:`.aiterate() <.Application.aiterate()>`
    * :py:meth:`.arun() <.Application.arun()>`
    * :py:meth:`.astream_result() <.Application.astream_result()>`
 
-2. For prototyping applications or high stakes operations,... go with sync.Check out:
+2. Use the synchronous interfaces otherwise -- when you have high CPU-bound applications (running models locally), or do not have async APIs to use:
 
    * :py:meth:`.build() <.ApplicationBuilder.build()>`
    * :py:meth:`.iterate() <.Application.iterate()>`
    * :py:meth:`.run() <.Application.run()>`
    * :py:meth:`.stream_result() <.Application.stream_result()>`
 
+Comparison
+----------
+
+A synchronous application processes tasks sequentially for every thread/process it executes, blocking entirely on the result
+of the prior call. For Burr, this means that two separate applications have to run in a separate thread/process, and that running
+parallel processes have to launch in a multithreaded/multi-processing environment, run, and join the results.
+
+In the case of Burr, this means that you have a 1:1 app -> thread/process mapping (unless you're using :ref:`parallelism <parallelism>` and explicitly multithreading sub-actions).
 
-Burr does both sync and async
-------------------------------
-Bellow we give a short breakdown when to consider each case.
+An asynchronous application can parallelize multiple I/O bound tasks within the confines of a single thread. At the time that
+a task blocks on I/O it can give control back to the process, allow it to run other tasks simultaneously.
+
+In the case of Burr, this allows you to run more than one applications,in parallel, on the same thread.
+Note, however, that you have to ensure the application is async all the way down -- E.G. that every blocking call
+is called using `await` -- if it blocks the event loop through a slow, synchronous call, it will block *all* current
+applications.
```

Comment:
```suggestion
In the case of Burr, Burr supports this model in running multiple Burr applications, in parallel, on the same as thread (i.e. the asyncio event loop).
Note, however, that you have to ensure your Burr application is async all the way down -- E.G. that every blocking call
is called using `await` -- if it blocks the event loop through a slow, synchronous call, it will block *all* current
applications.
```